report/4.Solution/GeneratedCppCode.tex

```diff
@@ -0,0 +1,87 @@
+% !TeX root = ../main.tex
+
+\subsection{Generated C++ code}
+
+While the previous section primarily focused on the code generation from an abstract point of view, this section will go into depth about the generated C++ code.
+Since C++ is the output language of the HCL compiler, all the language concepts of HCL has to somehow be represented in valid C++ code.
+
+\textbf{Primitives}
+
+The primitives of HCL includes numbers, text and booleans. 
+Booleans are straightforward, as HCL's boolean behaves exactly like the C++ boolean type.
+This means that the regular boolean type can be used for output type of booleans.
+
+As HCL aims for simplicity, only one number type was introduced. 
+However this type should also be able to store non-integers. 
+Therefore the simple solution was to turn all HCL numbers into C++'s double type.
+The solution with doubles is not ideal, but it will fulfill the requirements of the number type in HCL.
+
+C++ does have an implementation of a string type in its standard library.
+However, the Arduino does not include this implementation.
+The Arduino uses its own string implementation that varies from the standard implementation.
```

Comment:
Remove "That varies from the standard implementation" and add a "." after "implementation".
It is quite obvious that the custom implementation would differ from the standard implementation.